nengo_gui/static/datastore.js

```diff
@@ -146,3 +139,177 @@ Nengo.DataStore.prototype.get_last_data = function() {
     }
     return shown;
 }
+
+
+Nengo.VariableDataStore = function(dims, sim, synapse){
+    Nengo.DataStore.call(this, dims, sim, synapse);
+    this._dims = dims;
+    this._old_dims = dims;
+
+    Object.defineProperty(this, "dims", {
+        get: function(){
+            return this._dims;
+        },
+        set: function(dim_val){
+            // throw a bunch of errors if bad things happen
+            // assuming you can only grow dims and not shrink them
+            if(this._dims < dim_val){
+                for (var i=0; i < dim_val - this._dims; i++) {
+                    this.data.push([]);
+                }
+            } else if(this._dims > dim_val) {
+                console.log("OH HELL NO");
```

Comment:
This should be something informative ...
Would it be better to raise an exception?